lib/livebook_web/live/session_live.ex

```diff
@@ -1059,7 +1060,12 @@ defmodule LivebookWeb.SessionLive do
   end
 
   def handle_params(%{"tab" => tab}, _url, socket) when socket.assigns.live_action == :export do
-    {:noreply, assign(socket, tab: tab)}
+    has_stale_cell? =
+      socket.assigns.data_view.section_views
+      |> Enum.flat_map(& &1.cell_views)
+      |> Enum.any?(&(&1.eval.validity == :stale))
```

Comment:
```suggestion
    any_stale_cell? = any_stale_cell?(socket.private.data)
```

`data_view` is specifically for rendering, otherwise we want to compute things based on the original `data`:

```elixir
defp any_stale_cell?(data) do
  data.notebook
  |> Notebook.evaluable_cells_with_section()
  |> Enum.any?(fn {cell, _section} ->
    data.cell_infos[cell.id].eval.validity == :stale
  end)
end
```